### PR TITLE
New Cassandra Cheat Sheet

### PR DESCRIPTION
**Pull request by gabhi**
https://duck.co/ia/view/cassandra_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=cassandra cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/cassandra_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=cassandra cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/cassandra_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=cassandra cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/cassandra_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=cassandra cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/cassandra_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/cassandra.json [hunk 1]
```diff
@@ -0,0 +1,105 @@
+{
+  "id": "cassandra_cheat_sheet",
+  "name": "Cassandra",
+  "description": "A quick reference of Apache's Cassandra database",
+  "template_type": "terminal",
+  "aliases":["cqlsh","cassandra database", "datastax"],
+  "metadata": {
+    "sourceName": "GitHub",
+    "sourceUrl": "http://gabhi.github.io/cassandra_cheat_sheet/"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/cassandra.json]: Use `https`.

File: share/goodie/cheat_sheets/json/cassandra.json [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    "KEYSPACE",
+    "USER",
+    "TABLE",
+    "INDEX"
+  ],
+  "sections": {
+    "KEYSPACE": [
+      {
+        "key": "CREATE ( KEYSPACE | SCHEMA )  keyspace_name WITH REPLICATION = map AND DURABLE_WRITES = ( true | false )",
+        "val": "Define a new keyspace and its replica placement strategy."
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/cassandra.json]: We can lose the periods at the ends of the values.

Comment (mintsoft) [inline share/goodie/cheat_sheets/json/cassandra.json]: also probably worth indicating that `map` is a placeholder for a map of values, not the word map

File: share/goodie/cheat_sheets/json/cassandra.json [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      {
+        "key": "ALTER USER user_name WITH PASSWORD 'password' ( NOSUPERUSER | SUPERUSER )",
+        "val": "Alter existing user options."
+      },
+      {
+        "key": "DROP USER user_name",
+        "val": "Remove a user."
+      },
+      {
+        "key": "LIST permission_name PERMISSION| ( LIST ALL PERMISSIONS ) ON resource OF user_name NORECURSIVE",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/cassandra.json]: not really sure what this is supposed to represent; should it be:
`(PERMISSION| LIST ALL PERMISSIONS )` ?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=cassandra cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/cassandra_cheat_sheet_mobile.png)

**Comment by gabhi** [comment]
Thank you all for reviewing this pull request. I have fixed all the review comments. Kindly re-review this PR

File: share/goodie/cheat_sheets/json/cassandra.json [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      },
+      {
+        "key": "LIST USERS",
+        "val": "List existing users and their superuser status"
+      }
+
+    ],
+    "TABLE": [
+      {
+        "key": "CREATE TABLE keyspace_name.table_name ( column_definition, column_definition, ...) WITH property AND property ...",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/cassandra.json]: I'd probably take `keyspace_name` out of these as usually people `USE keyspace;`

File: share/goodie/cheat_sheets/json/cassandra.json [hunk 5]
```diff
@@ -0,0 +1,105 @@
+{
+  "id": "cassandra_cheat_sheet",
+  "name": "Cassandra",
+  "description": "A quick reference of Apache's Cassandra database",
+  "template_type": "terminal",
+  "aliases":["cqlsh","cassandra database", "datastax"],
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/cassandra.json]: Can you add `CQL` to these triggers

**Comment by gabhi** [comment]
@mintsoft Thank you for reviewing this PR. I have fixed and updated the PR.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=cassandra cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/cassandra_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=cassandra cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/cassandra_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/cassandra.json [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+      {
+        "key": "CREATE ( KEYSPACE | SCHEMA )  keyspace_name WITH DURABLE_WRITES = ( true | false )",
+        "val": "Define a new keyspace and its replica placement strategy"
+      },
+      {
+        "key": "USE keyspace_name",
+        "val": "Connect the client session to a keyspace."
+      },
+      {
+        "key": "ALTER ( KEYSPACE | SCHEMA ) keyspace_name WITH REPLICATION = map| ( WITH DURABLE_WRITES = ( true | false )) AND ( DURABLE_WRITES = ( true | false))",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/cassandra.json]: I don't under stand the significance of the `|` after map?

File: share/goodie/cheat_sheets/json/cassandra.json [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+      }
+
+    ],
+    "TABLE": [
+      {
+        "key": "CREATE TABLE table_name ( column_definition, column_definition, ...) WITH property AND property ...",
+        "val": "Define a new table"
+      },
+      {
+        "key": "ALTER TABLE table_name instruction",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/cassandra.json]: I think it'd be good to have a list of instructions on this cheatsheet too

File: share/goodie/cheat_sheets/json/cassandra.json [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+      },
+      {
+        "key": "LIST USERS",
+        "val": "List existing users and their superuser status"
+      }
+
+    ],
+    "TABLE": [
+      {
+        "key": "CREATE TABLE table_name ( column_definition, column_definition, ...) WITH property AND property ...",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/cassandra.json]: I think it'd be good to have an explanation of `column_definition` on this cheatsheet; especially the available types

File: share/goodie/cheat_sheets/json/cassandra.json [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+      {
+        "key": "SELECT select_expression FROM table_name WHERE relation AND relation ... ORDER BY ( clustering_column ( ASC | DESC )...) LIMIT n ALLOW FILTERING",
+        "val": "Retrieve data from a Cassandra table"
+      },
+      {
+        "key": "UPDATE table_name USING option AND option SET assignment, assignment, ... WHERE row_specification",
+        "val": "Update columns in a row"
+      },
+      {
+        "key": "DELETE column_name, ... | ( column_name term )  FROM table_name USING TIMESTAMP integer WHERE row_specification",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/cassandra.json]: Again I don't understand what this `|` is all about

File: share/goodie/cheat_sheets/json/cassandra.json [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+        "val": "Removes entire rows or one or more columns from one or more rows"
+      },
+      {
+        "key": "TRUNCATE table_name",
+        "val": "Remove all data from a table"
+      },
+      {
+        "key": "INSERT INTO table_name ( column_name, column_name...) VALUES ( value, value ... ) USING option AND option",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/cassandra.json]: Be nice to have a list of options too

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=cassandra cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/cassandra_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=cassandra cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/cassandra_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=cassandra cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/cassandra_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=cassandra cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/cassandra_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=cassandra cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/cassandra_cheat_sheet_mobile.png)

**Comment by gabhi** [comment]
@mintsoft I have fixed all the review comments. Kindly re-review

File: share/goodie/cheat_sheets/json/cassandra.json [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+      {
+        "key": "CREATE KEYSPACE keyspace_name WITH DURABLE_WRITES = ( true | false )",
+        "val": "Define a new keyspace and its replica placement strategy"
+      },
+      {
+        "key": "USE keyspace_name",
+        "val": "Connect the client session to a keyspace."
+      },
+      {
+        "key": "ALTER KEYSPACE keyspace_name WITH REPLICATION = map| ( WITH DURABLE_WRITES = ( true | false )) AND ( DURABLE_WRITES = ( true | false))",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/cassandra.json]: space after map?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=cassandra cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/cassandra_cheat_sheet_mobile.png)

**Comment by mintsoft** [comment]
@gabhi Hmm, the description bits look a bit crap: https://beta.duckduckgo.com/?q=cassandra+cheat+sheet&ia=cheatsheet&iax=1

**Comment by gabhi** [comment]
@mintsoft are you referring to the description for create table/alter table commands?

**Comment by gabhi** [comment]
I just checked cheat sheet for sql https://beta.duckduckgo.com/?q=sql+cheat+sheet&ia=answer&iax=1 which doesnt provide lot of options. Do you suggest that should do the same?

**Comment by mintsoft** [comment]
@gabhi personally I think this section doesn't look very good:

terrible

the SQL one is smaller and so is more readible. Could we break it up a little maybe?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=cassandra cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/cassandra_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
@gabhi Thanks! This LGTM :+1:

**Comment by daxtheduck** [comment]
@gabhi this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Comment by gabhi** [comment]
@moollaza thanks :-)

**Pull request merged**